tsdb/db_test.go

```diff
@@ -3113,80 +3113,15 @@ func TestNoPanicOnTSDBOpenError(t *testing.T) {
 		require.NoError(t, os.RemoveAll(tmpdir))
 	})
 
-	absdir, err := filepath.Abs(tmpdir)
-	require.NoError(t, err)
-	// Taking the file lock will cause TSDB startup error.
-	lockf, _, err := fileutil.Flock(filepath.Join(absdir, "lock"))
+	// Taking the lock will cause a TSDB startup error.
+	l, err := NewLocker(tmpdir, "tsdb", log.NewNopLogger(), nil)
 	require.NoError(t, err)
+	require.NoError(t, l.Lock())
 
 	_, err = Open(tmpdir, nil, nil, DefaultOptions(), nil)
 	require.Error(t, err)
 
-	require.NoError(t, lockf.Release())
-}
-
-func TestLockfileMetric(t *testing.T) {
```

Comment:
We should still be having this test to make sure that the locker is being used properly by the DB.

Reply:
Sure. I created some test utils to assert the general behavior here and reused it between DirLocker/TSDB/agent.